### PR TITLE
Include mixEnv parameter and apply to environment

### PR DESCRIPTION
**Pull request by mauricerkelly**
In a comment on closed issue #80, I made a suggestion that a workaround could be to apply the `mix compile` command to a different Mix environment so that a separation between the linter and IEx can be maintained.

For example, in normal usage, launching IEx will often be run in the `dev` environment. As a result, the build output will go to `_build/dev/`. When the linter runs, it outputs to the same location, causing confusion for IEx when it is ordered to recompile.

This change allows the user to specify a Mix environment that the linter should use. This allows linting for the `test` or `prod` environments if that is preferred, or allows a separation between IEx output and linter output.

Note: I had trouble getting linter-elixirc to load in `--dev` mode (complained about not being able to find the module atom-elixir). I did test this by applying changes directly to my regular package directory, but I'll keep trying to test it directly from dev mode. I just wanted to get this PR up for discussion.

### CODE REVIEW

**Comment by Arcanemagus** [comment]
This project doesn't have anything in it referencing an `atom-elixir`, maybe you have [atom-elixir](https://atom.io/packages/atom-elixir) installed as well?

File: README.md [hunk 1]
```diff
@@ -40,6 +40,8 @@ Plugin should work with default settings. If not:
 
 5.  `always use elixirc` option - leave it disabled, unless `mix compile` is too slow.
 
+6.  `mix env` option - leave it at the default (dev), unless you want to lint in a specific environment. This can also be useful if changes made in Atom are not being picked up in IEx.
```
Comment (Arcanemagus) [inline README.md]: How about this?:

Allows changing the build environment for lint runs. If using IEx at the same time as Atom this needs to be changed to allow IEx to pick up code changes.

**Comment by Arcanemagus** [comment]
Is there any downside to setting this to `lint` or something by default?

File: README.md (hunk 1 above)
Comment (mauricerkelly) [inline README.md]: I'm cool with that!

**Comment by mauricerkelly** [comment]
> This project doesn't have anything in it referencing an atom-elixir

Sorry - it was supposed to read `atom-linter`.

> Is there any downside to setting this to lint or something by default?

That's a great idea - I did a quick test and Mix seems to support arbitrary environment names, so it should work.

**Comment by Arcanemagus** [comment]
> Sorry - it was supposed to read atom-linter.

Ah, did you run `npm install` in the source directory?

> That's a great idea - I did a quick test and Mix seems to support arbitrary environment names, so it should work.

Glad to hear it works, my only concerns are:
* Additional space requirements that might introduce for the build artefacts
* Majorly increased initial compile time for the first lint build (?)

**Comment by mauricerkelly** [comment]
What if we set the default to be `test` as originally suggested? Devs are likely to use `test` anyway (if they're not testing, then they're naughty devs!) They're also less likely to run IEx in the `test` environment than they are to run in `dev` or `prod`. But having the configuration means they can choose their own configuration (e.g. `lint`) if they do have IEx conflicts, or simply have a non-standard setup.

BTW, `npm install` helped, but now the `linter` package is failing to launch because it can't find `minimatch`. Attempting to resolve.

**Comment by mauricerkelly** [comment]
I've made an update to the README as suggested - does this project prefer including all the commits, or squashing them?

**Comment by Arcanemagus** [comment]
> What if we set the default to be test as originally suggested?

Like I've said before, I don't use Elixir so whatever makes sense in its ecosystem is best 😉.

> does this project prefer including all the commits, or squashing them?

Depends, if people sign their commits I always merge, otherwise I usually merge if the commits make sense to be separate, or just squash and merge if they were just bugfixes.

tl;dr Do whatever works for you and I'll either merge or squash and merge 😛

**Comment by mauricerkelly** [comment]
Taking some time to think about it, I think the best approach is to stick with a default of `dev` for the environment, and let people update it as they see fit.

Still having trouble with my own Atom dev environment -- the `npm install` pulled down a load of packages, but Atom is still having trouble loading `minimatch` as a dependency of `atom-linter`. Any good ways to reset the environment and instructions for proper setup.

**Comment by mauricerkelly** [comment]
Nevermind. Got it sorted - before following your `npm install` advice I'd tried installing the `linter` package into `.atom/dev`. I guess that package was being loaded in preference to the version in the node_modules directory, and that package didn't have it's own node_modules directory!

So, I've been able to test properly, and can give it the full 👍

**Comment by mauricerkelly** [comment]
It's weird that CI is still running on this. Any way to kill the run and restart it?

**Comment by Arcanemagus** [comment]
It's actually a bug with CircleCI ([see here](https://discuss.circleci.com/t/github-status-wont-update-after-build-completes/10739/3)), the build status is green if you manually open it.

**Comment by mauricerkelly** [comment]
Thanks for accepting this PR - installing the update now!
